comp/forwarder/orchestrator/orchestratorimpl/forwarder_no_orchestrator.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build !orchestrator
+
+// Package orchestratorimpl implements the orchestrator forwarder component.
+package orchestratorimpl
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	"github.com/DataDog/datadog-agent/comp/forwarder/orchestrator"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newOrchestratorForwarder),
+)
+
+// newOrchestratorForwarder builds the orchestrator forwarder.
+// This func has been extracted in this file to not include all the orchestrator
+// dependencies (k8s, several MBs) while building binaries not needing these.
+func newOrchestratorForwarder(_ log.Component, _ config.Component, params Params) orchestrator.Component {
+	if params.UseNoopOrchestratorForwarder {
+		forwarder := optional.NewOption[defaultforwarder.Forwarder](defaultforwarder.NoopForwarder{})
+		return &forwarder
+	}
+	forwarder := optional.NewNoneOption[defaultforwarder.Forwarder]()
+	return &forwarder
```

Comment:
What's the difference between a noop forwarder and a empty optional ? Feels like both implement the same idea of "we don't have an orchestrator", no ?

Reply:
> What's the difference between a noop forwarder and a empty optional 

I keep the existing logic to not change the current behaviour.
It should be more or less the same but it is not easy to know if a `nil` forwarder (or `None` forwarder after my change) is strictly equivalent to Noop forwarder.